samples/cpp/text_generation/README.md

```diff
@@ -1,44 +1,146 @@
-# Text generation C++ greedy_causal_lm that supports most popular models like LLaMA 3
+# OpenVINO GenAI Text Generation Samples
 
-This example showcases inference of text-generation Large Language Models (LLMs): `chatglm`, `LLaMA`, `Qwen` and other models with the same signature. The application doesn't have many configuration options to encourage the reader to explore and modify the source code. For example, change the device for inference to GPU. The sample fearures `ov::genai::LLMPipeline` and configures it to run the simplest deterministic greedy sampling algorithm. There is also a Jupyter [notebook](https://github.com/openvinotoolkit/openvino_notebooks/tree/latest/notebooks/llm-chatbot) which provides an example of LLM-powered Chatbot in Python.
+These samples showcase the use of OpenVINO's inference capabilities for text generation tasks, including different decoding strategies such as beam search, multinomial sampling, and speculative decoding. Each sample has a specific focus and demonstrates a unique aspect of text generation.
+The applications don't have many configuration options to encourage the reader to explore and modify the source code. For example, change the device for inference to GPU.
+There are also Jupyter notebooks for some samples. You can find links to them in the appropriate sample descritions.
+
+## Table of Contents
+1. [Download and Convert the Model and Tokenizers](#download-and-convert-the-model-and-tokenizers)
+2. [Sample Descriptions](#sample-descriptions)
+3. [Troubleshooting](#troubleshooting)
+4. [Support and Contribution](#support-and-contribution)
 
 ## Download and convert the model and tokenizers
 
 The `--upgrade-strategy eager` option is needed to ensure `optimum-intel` is upgraded to the latest version.
 
-It's not required to install [../../export-requirements.txt](../../export requirements.txt) for deployment if the model has already been exported.
+It's not required to install [../../export-requirements.txt](../../export-requirements.txt) for deployment if the model has already been exported.
 
 ```sh
 pip install --upgrade-strategy eager -r ../../requirements.txt
 optimum-cli export openvino --trust-remote-code --model TinyLlama/TinyLlama-1.1B-Chat-v1.0 TinyLlama-1.1B-Chat-v1.0
 ```
 
-## Run
-
-Follow [Get Started with Samples](https://docs.openvino.ai/2024/learn-openvino/openvino-samples/get-started-demos.html) to run the sample.
-
-`greedy_causal_lm TinyLlama-1.1B-Chat-v1.0 "Why is the Sun yellow?"`
+Model examples to use for different samples:
+chat_sample - meta-llama/Llama-2-7b-chat-hf
+speculative_decoding_lm - meta-llama/Llama-2-13b-hf as main model and TinyLlama/TinyLlama-1.1B-Chat-v1.0 as draft model
+other samples - meta-llama/Llama-2-7b-hf
```

Comment:
it's drawn as plain text
![{76F1C33E-36E9-4076-9F20-B01069A8A810}](https://github.com/user-attachments/assets/7d799e25-7c97-4782-a92e-a16aec2aaac5)
which is hard to read

Reply:
updated